python/src/aac/context/definition_parser.py

```diff
@@ -813,11 +813,16 @@ def load_definitions(
             for definition in self.parsed_definitions + self.context.get_definitions()
             if definition.get_root_key() == "primitive"
         }
-        self.fully_qualified_name_to_definition = {
-            definition.get_fully_qualified_name(): definition
-            for definition in self.parsed_definitions + self.context.get_definitions()
-            if "package" in definition.structure[definition.get_root_key()]
-        }
+        self.fully_qualified_name_to_definition = {}
+        for definition in self.parsed_definitions + self.context.get_definitions():
+            if "package" in definition.structure[definition.get_root_key()]:
+                fully_qualified_name = definition.get_fully_qualified_name()
+
+                # This is so requirements specifically do not get overwritten.  Although other definition types will still get overwritten, so we may need to find a better solution eventually.
+                if definition.get_root_key() == "req":
```

Comment:
Need to do a similar evaluation with `req_spec` and be sure they aren't going to encounter a similar overwrite issue. May help shape what a "better solution" might need to look like if we confirm it is possible for other definitions to be overwritten. The original bug was around requirements, and both `req` and `req_spec` fall in that bucket.

Reply:
Im not really sure how I would do that.  req_spec does not have an ID or any other identifying fields other than its name.  Could we have a package field for req specs?